src/inventory.cpp

```diff
@@ -271,6 +271,30 @@ std::string ItemStack::getShortDescription(const IItemDefManager *itemdef) const
 	return desc;
 }
 
+f32 ItemStack::getRange(const IItemDefManager *itemdef) const
+{
+    std::string sRange = metadata.getString("range");
+    f32 range = -1;
+    if (sRange.empty())
+        range = getDefinition(itemdef).range;
+    if (!sRange.empty())
+    {
+        try {
+            //This is the best way (until otherwise said) to convert a string into a float.
+            //std::stof gives a compilation error along the lines of: namespace 'std' does not contain 'mystof'.
+			char *c = const_cast<char *>(sRange.c_str());
+			f32 _range = std::strtof(c, NULL);
```

Comment:
```suggestion
			f32 _range = std::strtof(sRange.c_str(), nullptr);
```
`const_cast` is a special feature and should be treated with caution and respect.